### PR TITLE
Jafin david 239 - sidebar

### PR DESCRIPTION
**Pull request by flecmen**

### CODE REVIEW

File: Page/PageWrapper.vue [hunk 1]
```diff
@@ -102,6 +107,10 @@ onMounted(() => {
   &.is-padded {
     --apply: m-t-$navHeight;
   }
+
+  &.is-mobile-and-sidebar-open {
+    --apply: overflow-x-hidden border-rd-l-2 p-r-0;
```
Comment (Kylyi) [inline Page/PageWrapper.vue:112]: we talked about this `rd` shit

File: Table/TableLayoutSettingsBtn.vue [hunk 2]
```diff
@@ -269,7 +269,7 @@ const $v = useVuelidate(
     color="ca"
     no-uppercase
     :label="$t('settings')"
-    label-class="hidden sm:block"
+    label-class="block"
```
Comment (Kylyi) [inline Table/TableLayoutSettingsBtn.vue:272]: Is this even necessary now?

File: Table/TableTop.vue [hunk 3]
```diff
@@ -386,51 +372,119 @@ function handleFitColumns() {
...
+          </span>
 
-        <template
-          v-if="config.table.useServerState && !tableTopFunctionality?.noLayout"
+          <!-- Columns btn -->
+          <TableColumnsBtn
+            v-if="!tableTopFunctionality?.noColumnSelection"
+            v-model:columns="columns"
+          />
+          <!-- Autofit btn -->
```
Comment (Kylyi) [inline Table/TableTop.vue:456]: separate with new line

File: Table/TableTop.vue [hunk 4]
```diff
@@ -386,51 +372,119 @@ function handleFitColumns() {
...
+            size="sm"
+            no-uppercase
+            icon="material-symbols:fit-width"
+            color="ca"
+            :label="$t('table.fitColumns')"
+            @click="handleFitColumns"
+          />
+        </div>
+        <div
+          flex
```
Comment (Kylyi) [inline Table/TableTop.vue:468]: Please use `flex="~ items-center"`

File: Table/TableTop.vue [hunk 5]
```diff
@@ -386,51 +372,119 @@ function handleFitColumns() {
...
+            <!-- Layout selector -->
+            <TableLayoutSelector
+              v-model:query-builder="queryBuilder"
+              m-auto
+            />
+
+            <!-- Layout settings -->
+            <TableLayoutSettingsBtn
+              :non-saveable-settings="nonSavableSettings"
+              label="Layout settings"
```
Comment (Kylyi) [inline Table/TableTop.vue:423]: i18n

File: Table/TableTop.vue [hunk 6]
```diff
@@ -406,17 +416,62 @@ function handleFitColumns() {
...
+            color="ca hover:negative"
+            border="2 transparent hover:negative"
+            data-cy="remove-filters"
+          >
+            <Menu
+              :placement="isMobile ? 'bottom' : 'left'"
+              hide-header
+              :no-arrow="false"
+              content-class="gap-1"
+            >
```
Comment (Kylyi) [inline Table/TableTop.vue:442]: Also, add comments for each section

File: Page/PageWrapper.vue [hunk 7]
```diff
@@ -71,25 +74,34 @@ onMounted(() => {
...
     </Component>
-    <slot v-if="!loading" />
 
-    <slot
-      v-else
-      name="loading"
+    <!-- Content -->
+    <ScrollArea
+      class="page-wrapper__content"
+      p="s-2! md:s-5! e-2 md:e-0"
```
Comment (Kylyi) [inline Page/PageWrapper.vue:81]: using `padding-inline-start` and `padding-inline-end` is a bit weird here... but okay :)

Comment (flecmen) [reply Page/PageWrapper.vue:81]: It is, but PerfectScrollbar uses it, and I couldnt override it with regular `padding`, so I adapted.

**Pull request merged**